astrolabe/runner.py

```diff
@@ -383,6 +383,28 @@ def __init__(self, *, client, admin_client, test_locator_token, configuration, x
         self.persist_clusters = persist_clusters
         self.no_create = no_create
         self.workload_startup_time = workload_startup_time
+        # Hardcoded to 12 hours, can be configurable in the future
+        self.project_expiration_threshold_seconds = 12 * 60 * 60
+
+        # Set up Atlas for tests.
+        # Step-1: ensure validity of the organization.
+        # Note: organizations can only be created by via the web UI.
+        org_id = self.config.organization_id
+        LOGGER.info("Verifying organization id: {!r}".format(org_id))
+        org = get_organization_by_id(
+            client=self.client, org_id=org_id)
+        LOGGER.info("Successfully verified organization {!r}".format(org.name))
+
+        # Step-2: purge old projects from organization.
```

Comment:
I think this ultimately would be better off as a separate build step after the tests complete.

Reply:
I can create it as a separate build step but I think it's more beneficial before the test runs.  Does that work with you @p-mongo ?